### PR TITLE
[SandboxIR] Implement AddrSpaceCastInst

### PR DESCRIPTION
**Pull request by vporpo**
This patch implements sandboxir::AddrSpaceCastInst which mirrors llvm::AddrSpaceCastInst.

### CODE REVIEW

File: llvm/include/llvm/SandboxIR/SandboxIR.h
Comment (tschuett) [inline llvm/include/llvm/SandboxIR/SandboxIR.h:1428]: ```Ì```  and  ```cast<Instruction>(From)``` are the same. Please use:
```
return I->getOpcode() == Opcode::AddrSpaceCast:
```

**Comment by tschuett** [comment]
You got approval for upstreaming SandboxIR. Your team opens PR, reviews, accepts, and merges them. You might promote somebody from your team to code owner.

**Review by tmsri — APPROVED** [review]
LGTM

File: llvm/include/llvm/SandboxIR/SandboxIR.h
Comment (vporpo) [reply llvm/include/llvm/SandboxIR/SandboxIR.h:1428]: Oops good catch.

**Pull request merged**
